library/src/main/java/pro/javacard/gp/GPSession.java

```diff
@@ -188,14 +187,14 @@ public static GPSession connect(APDUBIBO channel, AID sdAID) throws IOException,
      * Get the version and build information of the library.
      */
     public static String getVersion() {
-        try (InputStream versionfile = GPSession.class.getResourceAsStream("pro_version.txt")) {
-            String version = "unknown-development";
-            if (versionfile != null) {
-                try (BufferedReader vinfo = new BufferedReader(new InputStreamReader(versionfile, StandardCharsets.US_ASCII))) {
-                    version = vinfo.readLine();
-                }
+        Properties prop = new Properties();
+        try (InputStream versionfile = GPSession.class.getResourceAsStream("git.properties")) {
+            try {
+                prop.load(versionfile);
+                return prop.getProperty("git.commit.id.describe");
```

Comment:
Please have a default value (unknown-development?)

Reply:
I was under the impression that that the Properties class would throw an exception if the key is missing, rather than returning null.
My bad.